src/ucp/wireup/address.c

```diff
@@ -312,6 +312,11 @@ ucp_address_unpack_iface_attr(ucp_address_iface_attr_t *iface_attr,
     }
 }
 
+static inline int ucp_address_need_ep_addrs(ucp_ep_h ep)
```

Comment:
address.c should not handle "listener" logic. the caller of ucp_address_pack() should pass ep=NULL if ep addresses should not be used.